Content.Server/SprayPainter/SprayPainterSystem.cs

```diff
@@ -22,21 +35,80 @@ public override void Initialize()
         SubscribeLocalEvent<SprayPainterComponent, SprayPainterPipeDoAfterEvent>(OnPipeDoAfter);
 
         SubscribeLocalEvent<AtmosPipeColorComponent, InteractUsingEvent>(OnPipeInteract);
+
+        SubscribeLocalEvent<SprayPainterComponent, SprayPainterCanisterDoAfterEvent>(OnPaintableDoAfter);
+
+        SubscribeLocalEvent<SprayPainterComponent, AfterInteractEvent>(OnFloorAfterInteract);
+    }
+
+    private void OnFloorAfterInteract(Entity<SprayPainterComponent> ent, ref AfterInteractEvent args)
+    {
+        if (args.Handled || !args.CanReach)
+            return;
+
+        if (!args.ClickLocation.IsValid(EntityManager))
+        {
+            _popup.PopupEntity(Loc.GetString("spray-painter-invalid-location"), ent, args.User);
+            args.Handled = true;
+            return;
+        }
```

Comment:
is this needed? this just looks like a copypaste from the crayon system which seems to be the only place in the codebase to do this

Reply:
i'm assuming that's a yes? again, if i ask a question i would appreciate a reply instead of resolving silently